tests/check_check_master.c

```diff
@@ -455,23 +470,61 @@ START_TEST(test_check_failure_msgs)
     tr = tr_fail_array[i - passed];
     ck_assert_msg(tr != NULL, NULL);
     got_msg = tr_msg(tr);
-    expected_msg = master_tests[i].msg;
+    expected_msg = master_test->msg;
+
+    unsigned char not_equal = 0;
+
+    switch (master_test->msg_type) {
```

Comment:
No way. Regular expressions have their own syntax. Regular expression and string are different. Some symbols need to be escaped. Moreover, I think forward  and someday there could be one more `msg_type` for template substitution. It could be useful if some data is platform dependant and could be determined at the initialization. For example it could be data returned by the `localeconv` function.